src/vs/editor/contrib/referenceSearch/browser/referenceSearch.ts

```diff
@@ -191,31 +191,33 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 });
 
 
-export function provideReferences(model: editorCommon.IReadOnlyModel, position: Position): TPromise<Location[]> {
-
-	// collect references from all providers
-	const promises = ReferenceProviderRegistry.ordered(model).map(provider => {
-		return asWinJsPromise((token) => {
-			return provider.provideReferences(model, position, { includeDeclaration: true }, token);
-		}).then(result => {
-			if (Array.isArray(result)) {
-				return <Location[]>result;
-			}
-			return undefined;
-		}, err => {
-			onUnexpectedExternalError(err);
+export function provideReferences(model: editorCommon.IReadOnlyModel, position: Position): PPromise<void, Location[]> {
```

Comment:
`toggleWidget` requires that I pass in some object that contains the callback. The options I considered are
1. Use PPromise
2. Create a new object that merely holds the callback function
3. Refactor `toggleWidget`

I initially implemented (2) but tried (1) and thought it was cleaner. You would prefer (2)?

This is a situation where it would really be great to just have an observable type object: https://github.com/Microsoft/vscode/issues/28851